libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -99,7 +100,7 @@ internal class MapboxTripSession(
                 updateLegIndexJob?.cancel()
                 updateRouteProgressJob?.cancel()
 
-                threadController.getMainScopeAndRootJob().scope.launch {
+                threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
```

Comment:
Alternatively we can use an immediate dispatcher by default in `getMainScopeAndRootJob`.
@mapbox/navigation-android , what do you think?

Reply:
makes sense

Reply:
For a lot of the cases it definitely make sense but we'll need to double check if there aren't implementations that do rely on the call being posted back to the message queue, let's think about that separately from this PR.

Reply:
Yes, I fixed only path that affects test `addVoiceInstructionObserver` 😱

Reply:
will add